### PR TITLE
Ep9/14 invitation emails

### PR DESCRIPTION
**Pull request by dwyerse**
When you invite a new user with their email. A message will be sent to that email with a link to the response page.

### CODE REVIEW

File: routes/invitationResponse.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+		var index = inviteeList.findIndex(function( obj ) {
+			return obj.email == userEmail;
+		});
+		if (index !== -1) inviteeList.splice(index, 1);
+		inviteeList.push({email:userEmail, state:req.body.response});		
+		eventMapper.updateInviteeList(req.params.event_id,inviteeList,function(err){
+			if(err){
+				res.send(err);
+			}
+			res.redirect('/invitationResponse/view/' + +req.params.event_id +'/'+userEmail);
```
[the remark this answers is not in the document]
Comment (SeanDurban) [reply routes/invitationResponse.js]: Ah I've just figured that if you invite a person without an acc then that could cause issues

[the remark this answers is not in the document]
Comment (dwyerse) [reply routes/invitationResponse.js]: Yeah also if you weren't logged in that would also be an issue

File: mappers/eventMapper.js [hunk 2]
```diff
@@ -58,6 +58,23 @@ function updateEventDetailsBy_event_id(event_id,eventObj,callback){
 	});
 }
 
+function updateInviteeList(event_id,inviteeList,callback){
```
[the remark this answers is not in the document]
Comment (dwyerse) [reply mappers/eventMapper.js]: I didn't have that function when I wrote this.

[the remark this answers is not in the document]
Comment (SeanDurban) [reply mappers/eventMapper.js]: Wouldnt worry about that then. We can sort it out tomorrow separately

File: routes/invitationResponse.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (SeanDurban) [reply routes/invitationResponse.js]: Yeah ok, still bit awkward but think its better of the two.

**Pull request merged**